Display contracts (TG-190)

jdtzmn opened the pull request:
### Added

- Contract endpoint to service API
- Contract list user interface

### Changed

- Docker config to use a different horizon image tag

### Removed

- "Create an offer" button from the navbar

jdtzmn committed:
Add contract endpoint to service API

jdtzmn committed:
Add contract list UI

Adds contract list to the flocx-ui page and updates the docker config to
use different tags for the horizon image.

gagansk commented on flocx_ui/api/flocx.py:
```diff
@@ -94,3 +94,13 @@ def offer_get(request, offer_id):
+    :param request: HTTP request
+    :return: A list of contracts
+    """
+    r = get('/contract', token=request.user.token.id)
```
Might be minor, but from here onwards, let us start using meaningful variable names. 'result' would be a better option.

jdtzmn replied:
Sure thing! I was following the same format as ironic-ui, but more meaningful variable names for this makes sense.

gagansk commented on flocx_ui/static/dashboard/project/flocx/contract-list/contract-list.controller.js:
```diff
@@ -0,0 +1,61 @@
+      // Get the list of contracts from the flocx-market
+      flocx.getContracts().then(onGetContracts);
+    }
+
```
Please add comments for JS functions.

gagansk commented on flocx_ui/static/dashboard/project/flocx/contract-list/contract-list.controller.js:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    function filter (contract) {
+      if (ctrl.pane === 'expired') {
```
Use a global variable called 'expired' and use that for conditional statements. There are some talks that are going on in terms of identifying status. So, in case the status name is changed in the future, we will have to change only one statement. The ideal way would be to declare all the variable in a file and import that file whenever it is required.

gagansk commented on flocx_ui/static/dashboard/project/flocx/contract-list/contract-list.html:
```diff
@@ -0,0 +1,43 @@
+      <button
+        type="button"
+        class="btn"
+        ng-class="{ 'btn-default': ctrl.pane !== 'active', 'btn-primary': ctrl.pane === 'active' }"
```
Same comment. Let us avoid using 'magic vales'. 
[https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad](url)

gagansk commented on flocx_ui/static/dashboard/project/flocx/contract-list/contract-list.html:
```diff
@@ -0,0 +1,43 @@
+    <table class="table" style="margin-top: 16px;">
+      <thead>
+        <tr>
+          <th scope="col">Contract ID</th>
```
What do you think about this idea? By clicking on a contract, the owner of the node will be able to see the offer that was made and the user will be able to see the bid that was made. Of course, this is a feature that we have to worry about later once we have the basic system in place. But we have to provide a mechanism to show the offer or the bid that is tied to the contract. We might track this as an issue if you feel this makes sense.

jdtzmn replied:
I like this idea, but I believe that there are more important user stories to complete before this.

gagansk replied:
Yes. We will have to worry about it only when all the user stories that we already have onboard have been addressed.

gagansk reviewed (changes requested):
I have requested some minor changes. Also, I would like to get your inputs on one of the suggestions I have made.

jdtzmn committed:
Refactor strings in contract controller to module constants

This commit replaced the exact strings which were used in the contract
controller with constants that are located in the flocx module
declaration.

jdtzmn committed:
Increment patch version

gagansk reviewed (approved):
lgtm
